pkg/aws/client/client.go

```diff
@@ -18,108 +18,124 @@ limitations under the License.
 }
 
-func (ac *awsClient) DeleteSecurityGroup(input *ec2.DeleteSecurityGroupInput) (*ec2.DeleteSecurityGroupOutput, error) {
-	return ac.ec2Client.DeleteSecurityGroup(input)
+func (ac *awsClient) DeleteSecurityGroup(ctx context.Context, input *ec2.DeleteSecurityGroupInput,
+	optFns ...func(*ec2.Options)) (*ec2.DeleteSecurityGroupOutput, error) {
+	return ac.ec2Client.DeleteSecurityGroup(ctx, input, optFns...)
 }
 
-func (ac *awsClient) DeleteTags(input *ec2.DeleteTagsInput) (*ec2.DeleteTagsOutput, error) {
-	return ac.ec2Client.DeleteTags(input)
+func (ac *awsClient) DeleteTags(ctx context.Context, input *ec2.DeleteTagsInput,
+	optFns ...func(*ec2.Options)) (*ec2.DeleteTagsOutput, error) {
+	return ac.ec2Client.DeleteTags(ctx, input, optFns...)
 }
 
-func (ac *awsClient) RevokeSecurityGroupIngress(input *ec2.RevokeSecurityGroupIngressInput) (*ec2.RevokeSecurityGroupIngressOutput, error) {
-	return ac.ec2Client.RevokeSecurityGroupIngress(input)
+func (ac *awsClient) RevokeSecurityGroupIngress(ctx context.Context, input *ec2.RevokeSecurityGroupIngressInput,
+	optFns ...func(*ec2.Options)) (*ec2.RevokeSecurityGroupIngressOutput, error) {
+	return ac.ec2Client.RevokeSecurityGroupIngress(ctx, input, optFns...)
 }
 
-func (ac *awsClient) DescribeInstanceTypeOfferings(
-	input *ec2.DescribeInstanceTypeOfferingsInput) (*ec2.DescribeInstanceTypeOfferingsOutput, error) {
-	return ac.ec2Client.DescribeInstanceTypeOfferings(input)
+func (ac *awsClient) DescribeInstanceTypeOfferings(ctx context.Context, input *ec2.DescribeInstanceTypeOfferingsInput,
+	optFns ...func(*ec2.Options)) (*ec2.DescribeInstanceTypeOfferingsOutput, error) {
+	return ac.ec2Client.DescribeInstanceTypeOfferings(ctx, input, optFns...)
 }
 
 func New(accessKeyID, secretAccessKey, region string) (Interface, error) {
-	sess, err := session.NewSession(&aws.Config{
-		Credentials:      credentials.NewStaticCredentials(accessKeyID, secretAccessKey, ""),
-		Region:           aws.String(region),
-		EndpointResolver: endpoints.ResolverFunc(awsChinaEndpointResolver),
-	})
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")),
+		config.WithEndpointResolver(aws.EndpointResolverFunc(awsChinaEndpointResolver)))
 	if err != nil {
-		return nil, fmt.Errorf("failed to create aws session: %v", err)
+		return nil, fmt.Errorf("failed to load AWS configuration: %w", err)
 	}
 
 	return &awsClient{
-		ec2Client: ec2.New(sess),
+		ec2Client: *ec2.NewFromConfig(cfg),
 	}, nil
 }
 
-func awsChinaEndpointResolver(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-	if service != route53.EndpointsID || region != "cn-northwest-1" {
-		return endpoints.DefaultResolver().EndpointFor(service, region, optFns...)
+func awsChinaEndpointResolver(service, region string) (aws.Endpoint, error) {
```

Comment:
nit but this no longer resolves.

```suggestion
func newChinaEndpoint(service, region string) (aws.Endpoint, error) {
```

Reply:
I hesitated to make it anonymous initially, I’ve just done that instead.